include/spi_drv.h

```diff
@@ -49,15 +49,23 @@
 #include "hal/spi/spi_drv_nrf52.h"
 #endif
 
+#if defined(PLATFORM_nxp_p1021)
+#include "hal/spi/spi_drv_nxp.h"
+#endif
+
 void spi_init(int polarity, int phase);
 void spi_release(void);
 
-#if defined(SPI_FLASH) || defined(WOLFBOOT_TPM)
+#ifdef SPI_FLASH
 void spi_cs_on(uint32_t base, int pin);
 void spi_cs_off(uint32_t base, int pin);
 void spi_write(const char byte);
 uint8_t spi_read(void);
-#endif /* SPI_FLASH || WOLFBOOT_TPM */
+#endif
+
+#ifdef WOLFBOOT_TPM
+int spi_xfer(int cs, const uint8_t* tx, uint8_t* rx, uint32_t sz);
```

Comment:
Hrmmm.  I think we may need an interface that allows continued transactions where the CS does not get deserted, specifically when the total size of the transaction is unknown at the beginning, like with wait states.

Reply:
I agree and the eSPI has the ability to assert / dessert the CS separately.